doc/specs/stdlib_math.md

```diff
@@ -0,0 +1,101 @@
+---
+title: math
+---
+
+# The `stdlib_math` module
+
+[TOC]
+
+## Introduction
+
+`stdlib_math` module provides with general purpose mathematical functions.
+
+
+## Procedures and Methods provided
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### `clip` function
+
+#### Description
+
+Limits the input value `x` to the given interval [`xmin`, `xmax`] (interval is `xmin` and `xmax` inclusive). Returns a value which lies in the given interval and is closest to the input value `x`.  
```

Comment:
Are these to sentences not saying the same thimg?

Reply:
I just wanted to make it clearer by using the second sentence just saying that it limits felt a bit vague to me, so that's why i did that. If you wish I can remove it.

Reply:
may be I can use `i.e.` after first sentence to indicate that both sentences signify the same idea and second one is an explanation of the first.

Reply:
I think I would remove the first sentence. The second sentence is clearer.

Reply:
ok! will make the necessary changes in next PR.

Reply:
How about "Returns `xmin` if `x < xmin`, `xmax` if `x > xmax`, and `x` otherwise." ?

Reply:
I have made some changes, let me know if you there is something which still needs to be changed.